### PR TITLE
TA#21111 Add module web_trash_condition

### PR DESCRIPTION
**Pull request by ddufresne**

### CODE REVIEW

File: web_trash_condition/static/src/js/trash.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    License LGPL-3.0 or later (http://www.gnu.org/licenses/LGPL.html).
+*/
+odoo.define("web_trash_condition", function(require) {
+"use strict";
+
+const trashInvisibleField = "trash_widget_invisible";
+
+// The editable list properties must be included before
+// so that the method _renderRow can be overrided properly.
+require("web.EditableListRenderer")
```
Comment (ddufresne) [inline web_trash_condition/static/src/js/trash.js]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/manual/numigi/odoo-web-addons/pullRequest?prid=5175778)

File: web_trash_condition/static/src/js/trash.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+        return $row
+    },
+
+    _isTrashInvisible(record) {
+        return Boolean(record.data[trashInvisibleField]);
+    },
+
+    _hideTrashFromRow($row) {
+        $row.find('.o_list_record_remove button').addClass("d-none");
```
Comment (ddufresne) [inline web_trash_condition/static/src/js/trash.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/numigi/odoo-web-addons/pullRequest?prid=5175778)

File: web_trash_condition/static/src/js/trash.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+        if(this._isTrashInvisible(record)){
+            this._hideTrashFromRow($row);
+        }
+
+        return $row
+    },
+
+    _isTrashInvisible(record) {
+        return Boolean(record.data[trashInvisibleField]);
```
Comment (ddufresne) [inline web_trash_condition/static/src/js/trash.js]: Codacy Issue found: [Function Call Object Injection Sink](https://app.codacy.com/manual/numigi/odoo-web-addons/pullRequest?prid=5175778)

File: web_trash_condition/static/src/js/trash.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+require("web.ListRenderer").include({
+    _renderRow(record) {
+        var $row = this._super.apply(this, arguments);
+
+        if(this._isTrashInvisible(record)){
+            this._hideTrashFromRow($row);
+        }
+
+        return $row
```
Comment (ddufresne) [inline web_trash_condition/static/src/js/trash.js]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/manual/numigi/odoo-web-addons/pullRequest?prid=5175778)

**Review by foutoucour — APPROVED** [review]

**Pull request merged**